meson.build

```diff
@@ -0,0 +1,565 @@
+    radial_gradient_test = executable('radial_gradient_test', sources: [test_engine_src, 'tests/radial_gradient.c'], c_args: test_c_args, dependencies: test_engine_deps, include_directories: test_engine_includes, install: INSTALL_TEST, install_dir: get_option('bindir') / 'vkvg_tests/')
+    recording_test = executable('recording_test', sources: [test_engine_src, 'tests/recording.c'], c_args: test_c_args, dependencies: test_engine_deps, include_directories: test_engine_includes, install: INSTALL_TEST, install_dir: get_option('bindir') / 'vkvg_tests/')
+    rect_fill_test = executable('rect_fill_test', sources: [test_engine_src, 'tests/rect_fill.c'], c_args: test_c_args, dependencies: test_engine_deps, include_directories: test_engine_includes, install: INSTALL_TEST, install_dir: get_option('bindir') / 'vkvg_tests/')
+    save_restore_test = executable('save_restore_test', sources: [test_engine_src, 'tests/save_restore.c'], c_args: test_c_args, dependencies: test_engine_deps, include_directories: test_engine_includes, install: INSTALL_TEST, install_dir: get_option('bindir') / 'vkvg_tests/')
+    simple_paint_test = executable('simple_paint_test', sources: [test_engine_src, 'tests/simple_paint.c'], c_args: test_c_args, dependencies: test_engine_deps, include_directories: test_engine_includes, install: INSTALL_TEST, install_dir: get_option('bindir') / 'vkvg_tests/')
+    stroke_test = executable('stroke_test', sources: [test_engine_src, 'tests/stroke.c'], c_args: test_c_args, dependencies: test_engine_deps, include_directories: test_engine_includes, install: INSTALL_TEST, install_dir: get_option('bindir') / 'vkvg_tests/')
+    surface_test = executable('surface_test', sources: [test_engine_src, 'tests/surface.c'], c_args: test_c_args, dependencies: test_engine_deps, include_directories: test_engine_includes, install: INSTALL_TEST, install_dir: get_option('bindir') / 'vkvg_tests/')
+    svg_test = executable('svg_test', sources: [test_engine_src, 'tests/svg.c'], c_args: test_c_args, dependencies: test_engine_deps, include_directories: test_engine_includes, install: INSTALL_TEST, install_dir: get_option('bindir') / 'vkvg_tests/')
+    test1_test = executable('test1_test', sources: [test_engine_src, 'tests/test1.c'], c_args: test_c_args, dependencies: test_engine_deps, include_directories: test_engine_includes, install: INSTALL_TEST, install_dir: get_option('bindir') / 'vkvg_tests/')
+    text_test = executable('text_test', sources: [test_engine_src, 'tests/text.c'], c_args: test_c_args, dependencies: test_engine_deps, include_directories: test_engine_includes, install: INSTALL_TEST, install_dir: get_option('bindir') / 'vkvg_tests/')
+    transform_test = executable('transform_test', sources: [test_engine_src, 'tests/transform.c'], c_args: test_c_args, dependencies: test_engine_deps, include_directories: test_engine_includes, install: INSTALL_TEST, install_dir: get_option('bindir') / 'vkvg_tests/')
```

Comment:
You might want to just have this be a loop over an array.

```meson
testnames = ['arcs', 'bezier', ..., 'transform']

foreach t: testnames
    testprog = executable(t +'test',
        sources: [test_engine_src, 'tests/@0@.c'.format(t)],
        c_args: test_c_args,
        dependencies: test_engine_deps,
        include_directories: test_engine_includes,
        install: INSTALL_TEST,
        install_dir: get_option('bindir') / 'vkvg_tests/'
    )
    # TODO: implement this
    # test(t, testprog)
endforeach
```

Reply:
That a pretty sensible approach, I like it. I try to conventionally keep it verbose to ensure it's readable and keep distinct variable names for each executable variable, but yep, I've rewritten to apply your version of it instead. I updated the source code crediting you for the idea.

Also we can't run test for each test automatically which is what I think your TODO is referring to, because it basically brings up a window and then run various iterations over it which can take a significant amount of time to run all of the tests. I have brought up the idea for "Unit-Test" version of such test cases over at #120 .

Reply:
Yeah, I figured there might be a good reason you weren't running it there so intentionally stubbed out the logic as a comment. It's mainly there to show where you'd then refer back to `testprog`.

Reply:
Using fs.stem to insert the executable definition into a dict is a neat way to reuse it for later in cases where the leading directory differs -- although I guess at the moment you're not really using them elsewhere at the moment.

Reply:
Yup, it might be handy for unit testing later on, I just need to discuss with @jpbruyere on rewriting test cases to differentiate between 2 modes, one for visual check (bringing it up on window) and other for unit testing and that would be where executable variables come in play for defining test cases in meson.

Reply:
Went ahead and removed the commented out lines.